lib/elixir/lib/map.ex

```diff
@@ -845,7 +845,9 @@ defmodule Map do
       {nil, %{a: 1}}
 
   """
-  @spec get_and_update(map, key, (value -> {get, value} | :pop)) :: {get, map} when get: term
+  @spec get_and_update(map, key, (value -> {current_value, new_value :: value} | :pop)) ::
+          {current_value, map}
```

Comment:
:+1: for `current_value` here, as it is also called if the value is nil. I also like `new_value`!